CI: use 2.4.6, 2.5.5, 2.6.2

olleolleolle opened the pull request:
This PR updates the CI matrix.

  - also: Removes an old setting from Travis. See https://blog.travis-ci.com/2018-11-19-required-linux-infrastructure-migration

JonRowe commented:
Great, do you want to use the rake task to spin up a PR for each repo based on this? When they pass I'll merge (due to the sudo change which I know is supposed to be an old setting but not sure if it affects our build).

olleolleolle commented:
@JonRowe I'll make an attempt!

olleolleolle commented:
The attempt created commits, this is true, but since I'm not a member of the RSpec org, pushing the changes was not possible for the script.

<details>

```
rake travis:create_pr_with_updates
==================================================
# rspec-core
----------------------------------------

==================================================
# rspec-expectations
----------------------------------------

==================================================
# rspec-mocks
----------------------------------------

==================================================
# rspec-rails
----------------------------------------

==================================================
# rspec-support
----------------------------------------

==================================================
# rspec-core
----------------------------------------
git checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.
git pull --rebase
Already up to date.
Current branch master is up to date.

==================================================
# rspec-expectations
----------------------------------------
git checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.
git pull --rebase
Already up to date.
Current branch master is up to date.

==================================================
# rspec-mocks
----------------------------------------
git checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.
git pull --rebase
Already up to date.
Current branch master is up to date.

==================================================
# rspec-rails
----------------------------------------
git checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.
git pull --rebase
Already up to date.
Current branch master is up to date.

==================================================
# rspec-support
----------------------------------------
git checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.
git pull --rebase
Already up to date.
Current branch master is up to date.

==================================================
# rspec-core
----------------------------------------
git checkout -b update-travis-build-scripts-2019-04-01-for-master
Switched to a new branch 'update-travis-build-scripts-2019-04-01-for-master'
git add .
git commit -m 'Updated travis build scripts (from rspec-dev)'
[update-travis-build-scripts-2019-04-01-for-master 91396f18] Updated travis build scripts (from rspec-dev)
 9 files changed, 12 insertions(+), 13 deletions(-)

==================================================
# rspec-expectations
----------------------------------------
git checkout -b update-travis-build-scripts-2019-04-01-for-master
Switched to a new branch 'update-travis-build-scripts-2019-04-01-for-master'
git add .
git commit -m 'Updated travis build scripts (from rspec-dev)'
[update-travis-build-scripts-2019-04-01-for-master 2bcdfdad] Updated travis build scripts (from rspec-dev)
 9 files changed, 12 insertions(+), 13 deletions(-)

==================================================
# rspec-mocks
----------------------------------------
git checkout -b update-travis-build-scripts-2019-04-01-for-master
Switched to a new branch 'update-travis-build-scripts-2019-04-01-for-master'
git add .
git commit -m 'Updated travis build scripts (from rspec-dev)'
[update-travis-build-scripts-2019-04-01-for-master 57c1b45b] Updated travis build scripts (from rspec-dev)
 9 files changed, 12 insertions(+), 13 deletions(-)

==================================================
# rspec-rails
----------------------------------------
git checkout -b update-travis-build-scripts-2019-04-01-for-master
Switched to a new branch 'update-travis-build-scripts-2019-04-01-for-master'
./script/before_update_travis_build.sh
Backing up custom Travis config
./script/after_update_travis_build.sh
Restoring custom Travis config
git add .
git commit -m 'Updated travis build scripts (from rspec-dev)'
[update-travis-build-scripts-2019-04-01-for-master 02dd4089] Updated travis build scripts (from rspec-dev)
 8 files changed, 8 insertions(+), 8 deletions(-)

==================================================
# rspec-support
----------------------------------------
git checkout -b update-travis-build-scripts-2019-04-01-for-master
Switched to a new branch 'update-travis-build-scripts-2019-04-01-for-master'
git add .
git commit -m 'Updated travis build scripts (from rspec-dev)'
[update-travis-build-scripts-2019-04-01-for-master 2dddd6d] Updated travis build scripts (from rspec-dev)
 9 files changed, 12 insertions(+), 13 deletions(-)

==================================================
# rspec-core
----------------------------------------
fatal: remote error:
  You can't push to git://github.com/rspec/rspec-core.git
  Use https://github.com/rspec/rspec-core.git
Push failed, force? (y/n)
y
git push origin +update-travis-build-scripts-2019-04-01-for-master
fatal: remote error:
  You can't push to git://github.com/rspec/rspec-core.git
  Use https://github.com/rspec/rspec-core.git
rake aborted!
Command failed with status (128): [git push origin +update-travis-build-scrip...]
/Users/olle/opensource/rspec-dev/Rakefile:459:in `block in force_update'
/Users/olle/opensource/rspec-dev/Rakefile:46:in `block (2 levels) in each_project'
/Users/olle/opensource/rspec-dev/Rakefile:42:in `chdir'
/Users/olle/opensource/rspec-dev/Rakefile:42:in `block in each_project'
/Users/olle/opensource/rspec-dev/Rakefile:41:in `each'
/Users/olle/opensource/rspec-dev/Rakefile:41:in `each_project'
/Users/olle/opensource/rspec-dev/Rakefile:451:in `each_project_with_common_build'
/Users/olle/opensource/rspec-dev/Rakefile:455:in `force_update'
/Users/olle/opensource/rspec-dev/Rakefile:294:in `block (2 levels) in <top (required)>'
/Users/olle/.rvm/gems/ruby-2.6.1/gems/rake-12.0.0/exe/rake:27:in `<top (required)>'
Tasks: TOP => travis:create_pr_with_updates
(See full trace by running task with --trace)
```

</script>

I will apply some elbow grease and have a few PRs made.

JonRowe commented:
Thanks so much! Our script (for prs) should be modifiable to work with forks, if you felt like a pr for that it'd be fantastic!

benoittgt commented:
Thanks a lot @olleolleolle 🎉
